src/main.rs

```diff
@@ -48,13 +48,18 @@ struct App {
     ///
     /// This disables all online audit rules, and prevents zizmor from
     /// auditing remote repositories.
-    #[arg(short, long, env = "ZIZMOR_OFFLINE", group = "_offline")]
+    #[arg(short, long, env = "ZIZMOR_OFFLINE", 
+        conflicts_with_all = ["gh_token", "gh_hostname"])]
     offline: bool,
 
     /// The GitHub API token to use.
-    #[arg(long, env, group = "_offline")]
+    #[arg(long, env)]
     gh_token: Option<String>,
 
+    /// The GitHub API Server Hostname, if you are running GHE.
```

Comment:
Nitpick: slightly shorter and more generic (since this will also allow people to proxy the GH API, in addition to working with GHE 🙂)

```suggestion
    /// The GitHub API host to use.
```